### ISSUE TITLE
Warn on manual implementation of `.pop_if()`

### ISSUE THREAD

**Issue by Iurig** [rust-lang/rust-clippy#17619]
### What it does

Warns on manually checking any predicate with an if that fails on an empty vector/binary heap, then only popping from such vector/binary heap inside such if.

### Advantage

- Improves code clarity/conciseness

### Drawbacks

I am new to rust, so in all truth I do not know what the drawbacks would be.

### Example

```rust
    let mut v = vec![1, 2, 3];
    if !v.is_empty() && v[v.len() - 1] == 1 {
        v.pop();
    }
```

or

```rust
    let mut v = vec![1, 2, 3];
    if v.last() == Some(&1) {
        v.pop();
    }
```

Could be written as:

```rust
    let mut v = vec![1, 2, 3];
    v.pop_if(|x|*x == 1);
```

### Comparison with existing lints

I haven't found any lint that does such thing.

**Comment by Iurig** [comment]
This has already been done. My search wasn't thorough enough.
